src/main/scala/org/neo4j/spark/dataframe/Neo4jDataFrame.scala

```diff
@@ -95,6 +95,22 @@ object Neo4jDataFrame {
     sqlContext.createDataFrame(rowRdd, CypherTypes.schemaFromNamedType(schema))
   }
 
+  def toJava(x: Any): Any = {
+    import scala.collection.JavaConverters._
+    x match {
+      case y: scala.collection.MapLike[_, _, _] => 
```

Comment:
I don't think that Neo4j supports maps as properties. I added the conversion so that users would get an error message saying that it wasn't supported by the Neo4j driver, rather than a cryptic type conversion error message.

Reply:
@JakobWyatt-VGW and for the `SetLike`?

Reply:
@conker84 Same story there -  Neo4j only supports primitive types and arrays of primitive types as properties.
Again, did the full conversion so that neo4j would output a nice message and the user would know the problem.

Reply:
Do you think makes sense simplify this? I mean, if it's a `MapLike` we can leave it as-is, but for SetLike and other Iterables, we can try to coerce them to array, wdyt?

Reply:
I don't think the SetLike should be coerced to Array, here is my reasoning:

1. The user can already convert from a Set to an Array within Spark/Scala, before the dataframe enters the function (explicit conversion).
2. Sets and Arrays are fundamentally different. How will the Array be ordered?
3. If the Set is silently converted to an Array, and the user does not know this (and just assumes that Neo4j supports Sets), then they'll be confused and start asking questions like:
    - Why is it taking so long to search for an element?
    - Why am I allowed to add duplicate items?
Before realizing that it was converted to an array without them knowing.

In my opinion, the conversion from a Set to an Array should explicitly be done by the user, so that they know exactly what is going on.

Reply:
@JakobWyatt-VGW please reuse this code:
https://github.com/neo4j-contrib/neo4j-spark-connector/blob/eb552c4b8abb2d243874cd2030a8dfeb262a2ec4/src/main/scala/org/neo4j/spark/Executor.scala#L21-L24

and rebase the code so Travis now will work

Reply:
@conker84 Rather than reusing the existing Executor.toJava, should we just replace the function with the new toJava method?
This gives better error messages by the Neo4j connector if the user inputs an inappropriate property, and is a bit more user-friendly.

Reply:
Can you detail the two errors with a simple use case? IMHO I think that we don't need to try to convert properties not managed by Neo4j, moreover I can't understand the difference between this:
```
case y: Iterable[_] => 
        y.map { item: Any => toJava(item) } asJava
```

where `toJava(item)` gets the following case:

```
case _ => 
        x
```

and this:

```
case y: Seq[_] => y.asJava 
```

Can you explain the differences?

Reply:
Hi @conker84 
Here is the use case for the two errors (`spark-shell --jars neo4j-spark-connector-full-2.4.0-M8.jar --conf spark.neo4j.bolt.password=<password>`):
```scala
import spark.implicits._
import org.neo4j.spark.dataframe.Neo4jDataFrame

val df = Seq((Map("a" -> "b"))).toDF()
Neo4jDataFrame.createNodes(sc, df, ("example", Seq("value")))
```
With full conversion to Maps, the error message is:
`org.neo4j.driver.v1.exceptions.ClientException: Property values can only be of primitive types or arrays thereof`

However, without the full conversion to Maps, the error message is:
`org.neo4j.driver.v1.exceptions.ClientException: Unable to convert scala.collection.immutable.Map$Map1 to Neo4j Value.`

I know that personally, it took me a while to figure out why the second error message was occurring, whereas the first is much more clear.



The difference between those 2 code snippets (provided by you) is that the collection may be nested.
If it is a nested sequence (sequence of sequences for example), then the code I am proposing will recursively convert everything to a java sequence.
However, the existing code will only convert the top level of the sequence.
Both methods allow writing to the database on a valid sequence, however differ in the error message presented on an invalid sequence, as explained above.
On top of that, if Neo4j decides some time in the future to support richer property values, then my code will not need any modification to keep working. However, the existing code will.

I'm happy to go either way, just let me know and I can commit the changes.